source/routing/raptor_api.cpp

```diff
@@ -187,20 +187,59 @@ static void co2_emission_aggregator(pbnavitia::Journey* pb_journey){
 
 static void compute_durations(pbnavitia::Journey* pb_journey) {
     uint32_t total_walking_duration = 0;
+    uint32_t total_car_duration = 0;
+    uint32_t total_bike_duration = 0;
     for (const auto& section: pb_journey->sections()) {
         if ((section.type() == pbnavitia::STREET_NETWORK && section.street_network().mode() == pbnavitia::StreetNetworkMode::Walking) ||
         (section.type() == pbnavitia::TRANSFER && section.transfer_type() == pbnavitia::walking) ||
         (section.type() == pbnavitia::CROW_FLY && section.street_network().mode() == pbnavitia::StreetNetworkMode::Walking)) {
             total_walking_duration += section.duration();
         }
+
+        if (section.type() == pbnavitia::STREET_NETWORK) {
+            if (section.street_network().mode() == pbnavitia::StreetNetworkMode::Car) {
+                total_car_duration += section.duration();
+            } else if ((section.street_network().mode() == pbnavitia::StreetNetworkMode::Bike) ||
+                       (section.street_network().mode() == pbnavitia::StreetNetworkMode::Bss)) {
+                total_bike_duration += section.duration();
+            }
+        }
     }
     const auto ts_departure = pb_journey->sections(0).begin_date_time();
     const auto ts_arrival = pb_journey->sections(pb_journey->sections_size() - 1).end_date_time();
     pbnavitia::Durations* durations = pb_journey->mutable_durations();
     durations->set_walking(total_walking_duration);
+    durations->set_bike(total_bike_duration);
+    durations->set_car(total_car_duration);
     durations->set_total(ts_arrival - ts_departure);
 }
 
+static void compute_distances(pbnavitia::Journey* pb_journey) {
+    uint32_t total_walking_distance = 0;
+    uint32_t total_car_distance = 0;
+    uint32_t total_bike_distance = 0;
+    for (const auto& section: pb_journey->sections()) {
+        if ((section.type() == pbnavitia::STREET_NETWORK && section.street_network().mode() == pbnavitia::StreetNetworkMode::Walking) ||
+        (section.type() == pbnavitia::TRANSFER && section.transfer_type() == pbnavitia::walking) ||
+        (section.type() == pbnavitia::CROW_FLY && section.street_network().mode() == pbnavitia::StreetNetworkMode::Walking)) {
+            total_walking_distance += section.length();
+        }
+
+        if (section.type() == pbnavitia::STREET_NETWORK) {
```

Comment:
Same as durations. All that code is the same as compute_durations, maybe mutualise in a compute_metadata function or something like that.